eras/conway/impl/testlib/Test/Cardano/Ledger/Conway/Imp/RatifySpec.hs

```diff
@@ -223,6 +225,54 @@ paramChangeAffectsProposalsSpec =
       Map.member (GovPurposeId childGai) <$> getParameterChangeProposals `shouldReturn` True
       isDRepAccepted childGai `shouldReturn` False
 
+committeeMinSizeAffectsInFlightProposalsSpec ::
+  forall era.
+  ConwayEraImp era =>
+  SpecWith (ImpTestState era)
+committeeMinSizeAffectsInFlightProposalsSpec =
+  describe "CommitteeMinSize affects in-flight proposals" $ do
+    let setCommitteeMinSize n = modifyPParams $ ppCommitteeMinSizeL .~ n
+        submitATreasuryWithdrawal = do
+          rewardAccount <- registerRewardAccount
+          submitTreasuryWithdrawals [(rewardAccount, Coin 1_000)]
+    it "TreasuryWithdrawal fails to ratify due to an increase in CommitteeMinSize" $ do
+      (drepC, hotCommitteeC, _gpiCC) <- electBasicCommittee
+      setCommitteeMinSize 1
+      gaiPC <-
+        submitParameterChange SNothing $
+          emptyPParamsUpdate
+            & ppuCommitteeMinSizeL .~ SJust 2
+      submitYesVote_ (CommitteeVoter hotCommitteeC) gaiPC
+      submitYesVote_ (DRepVoter drepC) gaiPC
+      treasury <- getsNES $ nesEsL . esAccountStateL . asTreasuryL
+      gaiTW <- submitATreasuryWithdrawal
+      submitYesVote_ (CommitteeVoter hotCommitteeC) gaiTW
+      submitYesVote_ (DRepVoter drepC) gaiTW
+      isCommitteeAccepted gaiTW `shouldReturn` True
+      passNEpochs 2
+      -- The ParameterChange prevents the TreasuryWithdrawal from being enacted.
+      getLastEnactedParameterChange `shouldReturn` SJust (GovPurposeId gaiPC)
+      isCommitteeAccepted gaiTW `shouldReturn` False
+      currentProposalsShouldContain gaiTW
+      getsNES (nesEsL . esAccountStateL . asTreasuryL) `shouldReturn` treasury
+    it "TreasuryWithdrawal ratifies due to a decrease in CommitteeMinSize" $ do
+      (drepC, hotCommitteeC, _gpiCC) <- electBasicCommittee
+      modifyPParams $ ppGovActionLifetimeL .~ EpochInterval 10
+      treasury <- getsNES $ nesEsL . esAccountStateL . asTreasuryL
+      gaiTW <- submitATreasuryWithdrawal
+      submitYesVote_ (CommitteeVoter hotCommitteeC) gaiTW
+      submitYesVote_ (DRepVoter drepC) gaiTW
+      setCommitteeMinSize 2
+      isCommitteeAccepted gaiTW `shouldReturn` False
+      passNEpochs 2
+      getsNES (nesEsL . esAccountStateL . asTreasuryL) `shouldReturn` treasury
+      -- We do not enact the ParameterChange here because that does not pass
+      -- ratification as the CC size is smaller than MinSize
+      setCommitteeMinSize 1
```

Comment:
Instead of changing the parameter (since that is an impossible case) let's instead add more committee members to satisfy the `CommitteeMinSize` and ensure that it also triggers the `TreasuryWithdrawal` enactment